### ISSUE TITLE
Doesn't match files in subdirectories by default, doesn't match files relative to .gitignore file

### ISSUE THREAD

**Issue by tanzislam** [sadesyllas/explorer-excluded-files#5]
According to [the `.gitignore` documentation](https://git-scm.com/docs/gitignore):
- If I have this pattern in my workspace-level `.gitignore`: `*.o`, then Git ignores `.o` files in both the top-level directory and any subdirectories. (Git also behaves the same way if I specified it like: `**/*.o`. A non-trailing `**` means "match zero or more directories", while a trailing `/**` means "match all files and directories inside this directory".)
- Pattern-matching occurs downwards from the location of the `.gitignore` file in which the pattern occurs, but not upwards. So if I have `*.o` in my `a/.gitignore` file in my repo, then Git will ignore `a/hello.o` and `a/c/foo.o`, but not `b/bar.o`.

Neither of these behaviors are replicated in VSCode by this plugin.
- VSCode's `files.exclude` seems to always need patterns to start with `**/` in order for them to match files in subdirectories. (The plugin should cater for this by prepending each pattern with `**/` if it does not already begin with it.)
- The patterns from my provided `.gitignore` file are directly enlisted in `files.exclude` without regard to the location of the `.gitignore` file the pattern is from. The plugin should cater for this by prepending each pattern with the directory of the contributory `.gitignore` file, and a `/` (and then `**/` as above).

Finally, the `README.md` says:

>If you specify a **pattern** beginning with `file://`, e.g., `"file://.gitignore"`, the **pattern** is interpreted as a file...

This led me to believe that I can get the plugin to pick up all my subdirectory-level `.gitignore` files by having something like `"file://**/.gitignore"`. [I was wrong](https://github.com/sadesyllas/explorer-excluded-files/blob/5e5a693f3995a113fb4122627349b1c2ab5b6c1e/src/extension.ts#L222-L224); perhaps it would be good to rephrase it to avoid confusion:

>If you specify an **entry** beginning with `file://`, e.g., `"file://.gitignore"`, the **entry** is interpreted as a file...

**Comment by mscottnelson** [comment]
I agree, was also misled by 'pattern' phrasing here. Would love to see the feature!
